module/x/peggy/types/confirm.go

```diff
@@ -0,0 +1,135 @@
+package types
+
+import (
+	"encoding/hex"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+type Confirm interface{}
+
+var (
+	_ Confirm = &ConfirmBatch{}
+	_ Confirm = &ConfirmLogicCall{}
+	_ Confirm = &ValsetConfirm{}
+)
+
+// Route should return the name of the module
+func (msg ConfirmBatch) Route() string { return RouterKey }
+
+// Type should return the action
+func (msg ConfirmBatch) Type() string { return "confirm_batch" }
+
+// ValidateBasic performs stateless checks
+func (msg ConfirmBatch) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(msg.Orchestrator); err != nil {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, msg.Orchestrator)
+	}
+	if err := ValidateEthAddress(msg.EthSigner); err != nil {
+		return sdkerrors.Wrap(err, "eth signer")
+	}
+	if err := ValidateEthAddress(msg.TokenContract); err != nil {
+		return sdkerrors.Wrap(err, "token contract")
+	}
+	_, err := hex.DecodeString(msg.Signature)
+	if err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "Could not decode hex string %s", msg.Signature)
+	}
+	return nil
+}
+
+// GetSignBytes encodes the message for signing
+func (msg ConfirmBatch) GetSignBytes() []byte {
+	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(msg))
+}
```

Comment:
?? why are we using amino?

Reply:
No need to review the code yet, it still needs to be properly cleaned up

Reply:
🔪  AMINO 🔪 

![proto](https://i.imgflip.com/52vwzg.jpg)

Reply:
amino is now removed.